File: How_Many_Numbers_Are_Smaller_Than_the_Current_Number.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var smallerNumbersThanCurrent = function(nums) {
    let counter=0,result =[];

    for(let i=0;i<nums.length ; i++){
        counter =0;
        for(let j=0;j<nums.length;j++){
            if(nums[j]<nums[i] && i!=j){
                counter ++ ;
            }
        }
        result.push(counter);
        
    }
    return result;
};

File: main-One/how-many-numbers-are-smaller-than-the-current-number.py
from collections import Counter
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
      mp = {}
      result = []

      for i in nums:
          if i not in mp:
              mp[i] = 0
          mp[i] += 1

      cuml = 0

      for i in sorted(mp.keys()):
          mp[i], cuml = cuml, cuml + mp[i]
          
      for i in nums:
          result.append(mp[i])

      return result
            
        
                